langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiEmbeddingModel.java

```diff
@@ -46,32 +59,48 @@
  */
 public class VertexAiEmbeddingModel implements EmbeddingModel {
 
-    private static final int BATCH_SIZE = 250; // Vertex AI has a limit of up to 250 input texts per request
-
+    private static final int COMPUTE_TOKENS_MAX_INPUTS_PER_REQUEST = 2048;
+    private static final int EMBEDDING_MAX_INPUTS_PER_REQUEST = 250;
+    private static final int MAX_TOTAL_TOKENS_COUNT_PER_EMBEDDING_CALL = 20_000;
     private final PredictionServiceSettings settings;
+    private final LlmUtilityServiceSettings llmUtilitySettings;
     private final EndpointName endpointName;
     private final Integer maxRetries;
+    private final Integer maxBatchSize;
+    private final Integer maxTokensPerBatch;
 
     public VertexAiEmbeddingModel(String endpoint,
                                   String project,
                                   String location,
                                   String publisher,
                                   String modelName,
-                                  Integer maxRetries) {
+                                  Integer maxRetries,
+                                  Integer maxBatchSize,
+                                  Integer maxTokensPerBatch) {
+
+        this.endpointName = EndpointName.ofProjectLocationPublisherModelName(
+            ensureNotBlank(project, "project"),
+            ensureNotBlank(location, "location"),
+            ensureNotBlank(publisher, "publisher"),
+            ensureNotBlank(modelName, "modelName")
+        );
+
         try {
             this.settings = PredictionServiceSettings.newBuilder()
                     .setEndpoint(ensureNotBlank(endpoint, "endpoint"))
                     .build();
+
+            this.llmUtilitySettings = LlmUtilityServiceSettings.newBuilder()
+                    .setEndpoint(settings.getEndpoint())
+                    .build();
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
-        this.endpointName = EndpointName.ofProjectLocationPublisherModelName(
-                ensureNotBlank(project, "project"),
-                ensureNotBlank(location, "location"),
-                ensureNotBlank(publisher, "publisher"),
-                ensureNotBlank(modelName, "modelName")
-        );
+
         this.maxRetries = getOrDefault(maxRetries, 3);
+
+        this.maxBatchSize = getOrDefault(maxBatchSize, EMBEDDING_MAX_INPUTS_PER_REQUEST);
+        this.maxTokensPerBatch = getOrDefault(maxTokensPerBatch, MAX_TOTAL_TOKENS_COUNT_PER_EMBEDDING_CALL);
```

Comment:
minor: please use DEFAULT_{PARAM_NAME} naming for default constants?
e.g. `DEFAULT_MAX_TOKENS_PER_BATCH`

Reply:
OK.
Good convention.

I'll update the PR (maybe tomorrow) with the comments addressed. 
Thank you for the review.

Reply:
Done, both constants with defaults updated with the naming convention